src/v2/Apps/Conversation/Components/Conversation.tsx

```diff
@@ -24,7 +15,42 @@ interface ItemProps {
 
 const Item: React.FC<ItemProps> = props => {
   const { item } = props
-  if (item.__typename === "Artwork") {
+  const itemType = item.__typename
+
+  const getImage = item => {
+    if (itemType === "Artwork") {
+      return item.image?.url
+    } else if (itemType === "Show") {
+      return item.coverImage?.url
+    } else {
+      return null
+    }
+  }
+
+  const itemDetails = item => {
+    if (item.__typename === "Artwork") {
+      return [
+        <Sans size="4" weight="medium" color="white100">
+          {item.artistNames}
+        </Sans>,
+        <Sans size="2" color="white100">
+          {item.title} / {item.date}
+        </Sans>,
+      ]
+    } else if (item.__typename === "Show") {
+      const itemLocation = item.fair?.location?.city
+      return [
+        <Sans size="4" weight="medium" color="white100">
+          {item.fair.name}
+        </Sans>,
+        <Sans size="2" color="white100">
+          {itemLocation && `${itemLocation}, `} {item?.fair?.exhibitionPeriod}
+        </Sans>,
+      ]
+    }
+  }
+
+  if (itemType === "Artwork" || itemType === "Show") {
     return (
       <Link
         href={item.href}
```

Comment:
Also I'm getting a type error for this. Is there a way I can get around this?

<img width="947" alt="Screen Shot 2020-06-12 at 4 00 09 PM" src="https://user-images.githubusercontent.com/5201004/84541759-feb48680-acc5-11ea-87d3-6deea1f80722.png">

Reply:
Yeah! So, the error is happening because the return type from the query on this item is a union of `Artwork` or `Show`. Anytime there's a union, it has a `__typename` field with the name of the type... but that'll _also_ include an `%other` value that's intended to be used for unexpected values. In this case if you check for `%other` above this conditional and just don't render anything in that case, the type error should go away.

Reply:
When you say check for `%other` above do you mean within in this conditional block?

```js
if (itemType === "%other") {
    return null
  }else if (itemType === "Artwork" || itemType === "Show") {
    ...
  } else {
    ...
  }
}
```

Reply:
What I was also thinking is to define a type like `ItemType` that represents the union type we desire:

`type ItemType = "Artwork" | "Show"`

Then do something like:

```ts
if (item.__typename === "%other") return null
const itemType = item.__typename as ItemType
```

Where we cast `itemType` once we have done a check to returned null for any `%other` types.

Reply:
Yep, I think that works well. 👍